comp/core/secrets/secretsimpl/secrets.go

```diff
@@ -422,11 +448,62 @@ func (r *secretResolver) Refresh() (string, error) {
 		return "", err
 	}
 	b := new(strings.Builder)
-	err = t.Execute(b, refreshResult)
-	if err != nil {
+	if err = t.Execute(b, refreshResult); err != nil {
 		return "", err
 	}
-	return b.String(), nil
+	return b.String(), auditRecordErr
+}
+
+type auditRecord struct {
+	Handle string `json:"handle"`
+	Value  string `json:"value,omitempty"`
+}
+
+// addToAuditFile adds records to the audit file based upon newly refreshed secrets
+func (r *secretResolver) addToAuditFile(secretResponse map[string]string) error {
+	if r.auditFilename == "" {
+		return nil
+	}
+	if r.auditRotRecs == nil {
+		r.auditRotRecs = newRotatingNDRecords(r.auditFilename, config{})
+	}
```

Comment:
Why not initialize this in `Configure` ?

Reply:
It makes testing a little harder. Right now the tests get away with not needing to call `Configure`, but I'm not sure if that's a good thing or not.